ambry-router/src/main/java/com.github.ambry.router/GCMCryptoService.java

```diff
@@ -58,14 +62,30 @@ public GCMCryptoService(CryptoServiceConfig cryptoServiceConfig) {
       throw new IllegalArgumentException(
           "Unrecognized Encryption Decryption Mode " + config.cryptoServiceEncryptionDecryptionMode);
     }
+    fixedIvVal = new byte[ivValSize];
   }
 
   @Override
   public ByteBuffer encrypt(ByteBuffer toEncrypt, SecretKeySpec key) throws GeneralSecurityException {
+    return encrypt(toEncrypt, key, false);
+  }
+
+  /**
+   * Helper function to encrypt ByteBuffer with the given key. When useFixedIv is true, don't use a random iv byte
+   * array, using a all zero byte array instead. Only set it to be true in test.
+   * @param toEncrypt {@link ByteBuffer} that needs to be encrypted
+   * @param key the secret key (of type T) to use to encrypt
+   * @return the {@link ByteBuffer} containing the encrypted content. Ensure the result has all
+   * the information like the IV along with the encrypted content, in order to decrypt the content with a given key
+   * @throws {@link GeneralSecurityException} on any exception with encryption
+   */
+  ByteBuffer encrypt(ByteBuffer toEncrypt, SecretKeySpec key, boolean useFixedIv) throws GeneralSecurityException {
```

Comment:
Alternative, you can use `iv` as an argument. 
For test, pass in a fixedIv
For prod, pass in null. 

Depends on you. I am ok with both.

Reply:
I think I would prefer this way of passing in the `iv`. Then you could move the `fixedIvVal` constant out of this class and into the test class.